checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassJsonDetector.kt

```diff
@@ -0,0 +1,88 @@
+package com.kozaxinan.android.checks
+
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity.INFORMATIONAL
+import com.intellij.psi.PsiClass
+import org.jetbrains.uast.UAnnotated
+import org.jetbrains.uast.UField
+import org.jetbrains.uast.UMethod
+
+/**
+ * Check retrotif interface methods return type for SerializedName annotation.
+ */
+@Suppress("UnstableApiUsage")
+internal class NetworkLayerClassJsonDetector : RetrofitReturnTypeDetector() {
+
+  override fun createUastHandler(context: JavaContext) = NetworkLayerDtoFieldVisitor(context)
+
+  class NetworkLayerDtoFieldVisitor(private val context: JavaContext) : Visitor(context) {
+
+    override fun visitMethod(node: UMethod) {
+      val allFields = findAllFieldsOf(node)
+
+      val checkedFields = allFields.filterNot(::hasJsonNameAnnotation)
+          .map { it.name }
+      if (checkedFields.isNotEmpty()) {
+        context.report(
+            issue = ISSUE_NETWORK_LAYER_CLASS_JSON_RULE,
+            scopeClass = node,
+            location = context.getNameLocation(node),
+            message = "Return type doesn't have @Json annotation for $checkedFields fields."
+        )
+      }
+
+      val checkedClasses = allFields.mapNotNull { it.containingClass }
+          .toSet()
+          .filterNot(::hasJsonClassAnnotation)
+
+      if (checkedClasses.isNotEmpty()) {
+        context.report(
+            issue = ISSUE_NETWORK_LAYER_CLASS_JSON_CLASS_RULE,
+            scopeClass = node,
+            location = context.getNameLocation(node),
+            message = "Return type doesn't have @JsonClass annotation for $checkedClasses classes."
+        )
+      }
+    }
+
+    private fun hasJsonNameAnnotation(field: UField): Boolean {
+      return context
+          .evaluator
+          .getAllAnnotations(field as UAnnotated, true)
+          .mapNotNull { uAnnotation -> uAnnotation.qualifiedName }
+          .any { it.endsWith("Json") }
+    }
+
+    private fun hasJsonClassAnnotation(clazz: PsiClass): Boolean {
+      return clazz.annotations
+          .mapNotNull { uAnnotation -> uAnnotation.qualifiedName }
+          .any { it.endsWith("JsonClass") }
```

Comment:
do we care about create adapter true?

Reply:
it would be nice to have maybe, but I couldn't get the value in one try, so I didnt' want to spend more time on that. If you set it on `false` when you need it, your whole api doesn't work. If you set it on true for ENUMs, you get an error on compiling/assembling